packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -421,126 +423,103 @@ export const makeInnerVault = (
     );
   };
 
-  // Calculate the target level for Collateral for the vaultSeat and
-  // clientSeat implied by the proposal. If the proposal wants Collateral,
-  // transfer that amount from vault to client. If the proposal gives
-  // Collateral, transfer the opposite direction. Otherwise, return the current level.
-  const targetCollateralLevels = seat => {
-    const { vaultSeat } = state;
-    const proposal = seat.getProposal();
-    const startVaultAmount = getCollateralAllocated(vaultSeat);
-    const startClientAmount = getCollateralAllocated(seat);
-    if (proposal.want.Collateral) {
-      return {
-        vault: AmountMath.subtract(startVaultAmount, proposal.want.Collateral),
-        client: AmountMath.add(startClientAmount, proposal.want.Collateral),
-      };
-    } else if (proposal.give.Collateral) {
-      return {
-        vault: AmountMath.add(startVaultAmount, proposal.give.Collateral),
-        client: AmountMath.subtract(
-          startClientAmount,
-          proposal.give.Collateral,
-        ),
-      };
-    } else {
-      return {
-        vault: startVaultAmount,
-        client: startClientAmount,
-      };
-    }
-  };
+  // Was this simple, but you can't subtract `empty` from a missing allocation.
+  // const transfer = (fromSeat, toSeat, fromLoses, fromGains, key) => {
+  //   toSeat.incrementBy(fromSeat.decrementBy(harden({ [key]: fromLoses })));
+  //   fromSeat.incrementBy(toSeat.decrementBy(harden({ [key]: fromGains })));
+  // };
 
-  const transferCollateral = seat => {
-    const { vaultSeat } = state;
-    const proposal = seat.getProposal();
-    if (proposal.want.Collateral) {
-      seat.incrementBy(
-        vaultSeat.decrementBy(harden({ Collateral: proposal.want.Collateral })),
-      );
-    } else if (proposal.give.Collateral) {
-      vaultSeat.incrementBy(
-        seat.decrementBy(harden({ Collateral: proposal.give.Collateral })),
-      );
+  /**
+   * Stage a transfer between `fromSeat` and `toSeat`, specified as the delta between
+   * the gain and a loss on teh `fromSeat`. The gain/loss are typically from the
+   * give/want respectively of a proposal. The `key` is the allocation keyword.
+   *
+   * @param {ZCFSeat} fromSeat
+   * @param {ZCFSeat} toSeat
+   * @param {Amount} fromLoses
+   * @param {Amount} fromGains
+   * @param {Keyword} key
```

Comment:
consider putting `key` first because it's what's being transferred. "transfer Collateral from to"